### PR TITLE
interfaces: add ipmbdev

### PR DESCRIPTION
**Pull request by sergachev**
Add support of /dev/ipmb-X devices created by ipmb-dev-int driver on Linux.
Requires slave mode support by the I2C bus driver, list of such drivers can be seen this way:
https://elixir.bootlin.com/linux/latest/C/ident/i2c_slave_event

Tested on a Xilinx Zynq as Linux host with IPMB-0 bus of a NAT power supply.

### CODE REVIEW

File: pyipmi/interfaces/__init__.py [hunk 1]
```diff
@@ -16,12 +16,14 @@
...
 from .ipmitool import Ipmitool
 from .aardvark import Aardvark
+from .ipmbdev import IPMBDev
 from .mock import Mock
 from .rmcp import Rmcp
 
 INTERFACES = [
     Ipmitool,
     Aardvark,
+    IPMBDev,
```
Comment (mwalle) [inline pyipmi/interfaces/__init__.py:26]: IpmbDev

File: pyipmi/interfaces/ipmbdev.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        self.slave_address = slave_address
+        self.timeout = 0.25
+        self.max_retries = 3
+        self.next_sequence_number = 0
+
+        self._dev = os.open('/dev/ipmb-{}'.format(port), os.O_RDWR)
+
+    def raw_write(self, address, data):
+        # self._dev.i2c_master_write(address, data)
+        pass
```
Comment (mwalle) [inline pyipmi/interfaces/ipmbdev.py:28]: ?

File: pyipmi/ipmitool.py [hunk 3]
```diff
@@ -438,7 +438,7 @@ def usage(toplevel=False):
   -h               Show this help
   -v               Be verbose
   -V               Print version
-  -I <interface>   Set interface (available: rmcp, aardvark, ipmitool)
+  -I <interface>   Set interface (available: rmcp, aardvark, ipmitool, ipmbdev)
```
Comment (mwalle) [inline pyipmi/ipmitool.py:441]: theres a description for the parameters missing. the parameters  (ipmb device path) are missing in general ;)

File: README.rst [hunk 4]
```diff
@@ -22,6 +22,8 @@ Requirements
 ------------
 
 For IPMB interface a `Total Phase`_ Aardvark is needed.
+Another option is to use ipmb-dev-int driver on Linux with an I2C bus, driver of which supports slave mode:
+https://www.kernel.org/doc/html/latest/driver-api/ipmb.html
```
Comment (mwalle) [inline README.rst:27]: I think this supports a bit more documentation. I just had a quick glance at /dev/ipmb-N, and it seems to be a slave interface, but you just use it to do a request, right? This should be documented.

**Review by mwalle — COMMENTED** [review]
Thanks for the patch! Looks pretty good. See my comments and please add a SoB to your commit

File: README.rst (hunk 4 above)
Comment (sergachev) [reply README.rst:27]: Yes, it does support sending and receiving both requests and responses:
https://lists.ozlabs.org/pipermail/linux-aspeed/2019-November/002879.html

File: pyipmi/interfaces/__init__.py (hunk 1 above)
Comment (sergachev) [reply pyipmi/interfaces/__init__.py]: done

File: pyipmi/interfaces/ipmbdev.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+from ..logger import log
+from ..interfaces.ipmb import IpmbHeaderReq, checksum, rx_filter, encode_ipmb_msg
+
+
+class IPMBDev(object):
+    """This interface uses ipmb-dev-int linux driver."""
+
+    NAME = 'ipmbdev'
+
+    def __init__(self, slave_address=0x20, port=0):
```
[the remark this answers is not in the document]
Comment (sergachev) [reply pyipmi/interfaces/ipmbdev.py]: done

File: pyipmi/interfaces/ipmbdev.py (hunk 2 above)
Comment (sergachev) [reply pyipmi/interfaces/ipmbdev.py]: removed - raw_write() is present in aardvark but is not used anywhere at all

File: README.rst [hunk 6]
```diff
@@ -7,7 +7,7 @@ Features
 --------
 * native RMCP interface
 * legacy RMCP interface (using ipmitool)
-* IPMB interface (using the `Total Phase`_ Aardvark)
+* IPMB interface (using the `Total Phase`_ Aardvark or ipmb-dev-int on Linux)
```
[the remark this answers is not in the document]
Comment (sergachev) [reply README.rst]: sure, done

File: pyipmi/ipmitool.py (hunk 3 above)
Comment (sergachev) [reply pyipmi/ipmitool.py:441]: done

**Comment by hthiery** [comment]
thank you

**Pull request merged**